tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -3152,6 +3161,84 @@ def wait_on_pp_send_handles(self, send_handles, microbatch_id):
+        # disk onboards are refused there at KV-manager init).
+        tp_sync = (self.dist.tp_size > 1 and not self.enable_attention_dp
+                   and self.dist.pp_size == 1)
+        if tp_sync:
+            any_pending = self.dist.tp_allreduce(0 if all(local_ready) else 1,
+                                                 op=ReduceOp.MAX)
+            if not any_pending:
+                for req in ctx:
+                    req.py_onboard_pending = False
+                return
+            gathered = self.dist.tp_allgather(local_ready)
+            global_ready = [all(bits) for bits in zip(*gathered)]
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)